### ISSUE TITLE
UI: Sort and Group Apps by ID options

### ISSUE THREAD

**Issue by SteveTinkers** [celzero/rethink-app#1566]
1) It would be great to be able to sort applications by name, technical name, or application id. 
It's super convenient for managing system apps which share the same id.  It's nice to not have them all mixed up together in the list...  It would also be nice to see the ID and technical name for each... 

2) It would also be nice to have notes field for rules and apps
I sometimes can't remember remember why I allowed something.  Adding some notes for IPs, Domains, and Apps would be nice.

**Comment by ignoramous** [comment]
> It's nice to not have them all mixed up together in the list...

In the meanwhile, you can tap on the "filter" icon in the search box, and select *System*.

**Comment by opk12** [comment]
Cross-linking #1084 related to point 2.

**Comment by fernandovelarde559-maker** [comment]
I'm working on this

**Comment by ayeeeruben** [comment]
I'd like to work on this issue. Im partnered up with @fernandovelarde559-maker as part of the CTI and Codeday SOSE

My plan is to find where the Apps list is shown, add the new sorting options, and then work on adding notes for each app. I'll test each change as I go to make sure everything still works.

Does this sound like a good approach, or is there a better place to start?

Also, where should notes be implemented in the UI? I noticed an open space below the app details: 

<details><summary>screenshot</summary>

<img width="328" height="730" alt="Image" src="https://github.com/user-attachments/assets/d34eb45e-a6f5-46e4-8164-99dd65041fb3" />
</details>

Would this be a good spot to implement them?

@hussainmohd-a

**Comment by ayeeeruben** [comment]
> I'd like to work on this issue. Im partnered up with [@fernandovelarde559-maker](https://github.com/fernandovelarde559-maker) as part of the CTI and Codeday SOSE
> 
> My plan is to find where the Apps list is shown, add the new sorting options, and then work on adding notes for each app. I'll test each change as I go to make sure everything still works.
> 
> Does this sound like a good approach, or is there a better place to start?
> 
> Also, where should notes be implemented in the UI? I noticed an open space below the app details:
> 
> <img alt="Image" width="328" height="730" src="https://private-user-images.githubusercontent.com/125226745/618456719-d34eb45e-a6f5-46e4-8164-99dd65041fb3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODM0NjgyODUsIm5iZiI6MTc4MzQ2Nzk4NSwicGF0aCI6Ii8xMjUyMjY3NDUvNjE4NDU2NzE5LWQzNGViNDVlLWE2ZjUtNDZlNC04MTY0LTk5ZGQ2NTA0MWZiMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDcwN1QyMzQ2MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZTRmNzJhMDMyMWE0NzdkMDE5M2ZlMjdkYjMzNmRjNTY3YjcxYzI2OGNjMDMxOTBhMmQxYzQzMjM3YTJjMWM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.cHAmeMye56YtP6NLG-9QJoXqZknvD2xFR5o2QBIJA74">
> Would this be a good spot to implement them?
> 
> [@hussainmohd-a](https://github.com/hussainmohd-a)

Also, where should the technical names and ids go? Would this be placed alongside the "allowed" tags in the apps section or would this go in the app info page? I just need some clarity before starting to work on the issue.

**Comment by ignoramous** [comment]
Thanks for helping out.

> Also, where should notes be implemented in the UI? I noticed an open space below the app details:

Possibly as a "chip" named `Notes` next to the title "Firewall rules for this app"? Or, as an icon (with any appropriate MIT/Apache licensed icon from `icones.js.org`, for example) next to `i` at the top right hand corner of the UI?

The empty region circled out in the *app dashboard* screenshot above isn't always empty, especially for apps that actively connect to the Internet (check out any browser app's dashboard, for example, after loading a few websites).

> Also, where should the technical names and ids go? Would this be placed alongside the "allowed" tags in the apps section or would this go in the app info page?

What do you mean by *"allowed" tags*?

The app info page (app dashboard) already contains package name / technical name (which is `com.google.android.marvin.talkback` in the screenshot shared above) right below the actual app name (`Android Accessibility Suite`). Only the "UID" is missing, which I'm thinking can be shown right below the technical name (package name, really). UIDs on Android are integer values, so they shouldn't be long.

**Comment by ayeeeruben** [comment]
@hussainmohd-a This issue  is solved :)

**Comment by SteveTinkers** [comment]
Nice! Thanks! I am looking forward to using it!

**Comment by hussainmohd-a** [comment]
> [@hussainmohd-a](https://github.com/hussainmohd-a) This issue is solved :)

Thanks for the contribution :)
